ports/stm32/boards/stm32l152xe.ld

```diff
@@ -0,0 +1,39 @@
+/*
+    GNU linker script for STM32L152xE
+*/
+
+/* Specify the memory areas */
+MEMORY
+{
+    FLASH (rx)      : ORIGIN = 0x08000000, LENGTH = 512K /* entire flash */
+    FLASH_ISR (rx)  : ORIGIN = 0x08000000, LENGTH = 16K  /* sector 0-3 */
+    FLASH_FS (rx)   : ORIGIN = 0x08004000, LENGTH = 112K /* sectors 4-31 */
+    FLASH_TEXT (rx) : ORIGIN = 0x08020000, LENGTH = 384K /* sectors 32-127 */
```

Comment:
It looks like the L152 has uniform 512-byte erase page size (is that correct?).  So it would be simpler to put the filesystem at the end of flash, eg:
```
    FLASH (rx)      : ORIGIN = 0x08000000, LENGTH = 512K /* entire flash */
    FLASH_FS (rx)   : ORIGIN = 0x08064000, LENGTH = 112K /* sectors 912-1023 */
```
Then the board needs to specify `boards/common_basic.ld` in `mpconfigboard.mk`, and no need to specify `TEXT1_ADDR` (because the firmware is contiguous in flash, starting at `TEXT1_ADDR`).

Reply:
> It looks like the L152 has uniform 512-byte erase page size (is that correct?).

No, L152 has uniform 256-byte page size.
Also, L152 has uniform 4KB sectors (16 pages per sector).

> So it would be simpler to put the filesystem at the end of flash

I understand. I fixed it.